packages/fondue/src/components/Tooltip/Tooltip.tsx

```diff
@@ -71,39 +72,48 @@ export const Tooltip = ({
     zIndex = Z_INDEX_TOOLTIP,
     'data-test-id': dataTestId = 'fondue-tooltip',
 }: TooltipProps) => {
+    const timeoutRef = useRef<number | undefined>(undefined);
     const [open, setOpen] = useState(openOnMount);
-    const timeoutRef = useRef<ReturnType<typeof setTimeout> | null>(null);
     const [hasInteractiveElements, setHasInteractiveElements] = useState(false);
 
-    const handleHideTooltip = useCallback(() => {
-        handleTimeout(() => setOpen(false), leaveDelay, timeoutRef);
-    }, [disabled, leaveDelay, setOpen]);
-
-    const handleShowTooltip = useCallback(() => {
+    const handleShowTooltip = () => {
         if (!disabled) {
             handleTimeout(() => setOpen(true), enterDelay, timeoutRef);
         }
-    }, [disabled, enterDelay, setOpen]);
+    };
+
+    const handleHideTooltip = useCallback(() => {
+        handleTimeout(
+            () => {
+                setOpen(false);
+                window.clearTimeout(timeoutRef.current);
+            },
+            leaveDelay,
+            timeoutRef,
+        );
+    }, [leaveDelay]);
 
     const { dismissibleElementRef } = useClickOutside<HTMLDivElement>(handleHideTooltip, []);
 
     const focusAndMouseAttributes = {
         onBlur: handleHideTooltip,
         onFocus: handleShowTooltip,
```

Comment:
right now a PointerDevice can still activate the tooltip upon click. This is probably a UX-related question on whether we should disable it altogether or not? cc @noahwaldner @SamuelAlev 

It could be a controlled input since we have a `openOnMount` (which should be renamed since there is no tied logic to the lifecycle and it's just a prop passed by the consumer).

Reply:
Yes I would agree, just the click only happens on a mouse event, the click on mobile will not trigger it.
Its a good question if we want to trigger tooltip on a click or only on a hover.
If there is a  discussion how this tooltip should behave @floroz maybe we could discuss it and update it in another PR. What do you think?

Reply:
Yes, that sounds good to me 👍🏻